algorithms/src/snark/marlin/ahp/prover/constraint_system.rs

```diff
@@ -56,11 +70,34 @@ impl<F: Field> ConstraintSystem<F> {
         make_matrices_square(self, num_variables);
         assert_eq!(self.num_public_variables + self.num_private_variables, self.num_constraints, "padding failed!");
     }
+
+    fn lookup(&mut self, key: &[LinearCombination<F>], table_index: usize) -> Result<F, SynthesisError> {
```

Comment:
To confirm, this is a "native" lookup, right?

Reply:
yeah!